frontend/packages/dev-console/src/utils/__tests__/application-utils.spec.ts

```diff
@@ -0,0 +1,142 @@
+import * as k8s from '@console/internal/module/k8s';
+import {
+  ImageStreamModel,
+  ServiceModel,
+  DeploymentConfigModel,
+  RouteModel,
+  BuildConfigModel,
+  DaemonSetModel,
+  StatefulSetModel,
+} from '@console/internal/models';
+import {
+  RevisionModel,
+  ConfigurationModel,
+  ServiceModel as KnativeServiceModel,
+  RouteModel as KnativeRouteModel,
+} from '@console/knative-plugin';
+import * as utils from '@console/internal/components/utils';
+import { TopologyDataResources } from '../../components/topology/topology-types';
+import {
+  transformTopologyData,
+  getResourceDeploymentObject,
+} from '../../components/topology/topology-utils';
+import { cleanUpWorkload } from '../application-utils';
+import { MockResources } from '../../components/topology/__tests__/topology-test-data';
+import { MockKnativeResources } from '../../components/topology/__tests__/topology-knative-test-data';
+
+import Spy = jasmine.Spy;
+
+const spyAndReturn = (spy: Spy) => (returnValue: any) =>
```

Comment:
Can return value be typed?

Reply:
spyAndReturn is a generic test helper method, which can take in any spy and will resolve any value that is passed to it. IMO, we cannot bound this to any particular type